### PR TITLE
CXP-54: Access to the error list of a connection

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/GetErrorCountPerConnectionHandler.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        $this->selectErrorCountByConnectionQuery = $selectErrorCountByConnectionQuery;
+    }
+
+    /**
+     * @return ErrorCountPerConnection[]
+     */
+    public function handle(GetErrorCountPerConnectionQuery $query): array
+    {
+        $errorCountsByConnection = $this
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/GetErrorCountPerConnectionHandler.php:30]:
```suggestion
        $errorCountPerConnection = $this
```

Comment (pierrejolly) [reply src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/GetErrorCountPerConnectionHandler.php:30]: Oops I missed this one

File: src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/GetErrorCountPerConnectionHandler.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * @author Pierre Jolly <pierre.jolly@akeneo.com>
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class GetErrorCountPerConnectionHandler
+{
+    /** @var SelectErrorCountPerConnectionQuery */
+    private $selectErrorCountByConnectionQuery;
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/GetErrorCountPerConnectionHandler.php:18]:
```suggestion
    private $selectErrorCountPerConnectionQuery;
```

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCountPerConnection.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            $this->errorCounts[] = $errorCount;
+        }
+    }
+
+    public function normalize(): array
+    {
+        $errorCountPerConnection = [];
+
+        foreach ($this->errorCounts as $errorCount) {
+            $errorCountPerConnection[] = $errorCount->normalize();
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCountPerConnection.php:36]:
```suggestion
            $data = $errorCount->normalize();
            $errorCountPerConnection[$data['connection_code']] = (int) $data['count'];
```
We defined a format like this for the internal API response:
```json
{
  "code_1": 3,
  "code_2": 9
}
```

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalSelectErrorCountPerConnectionQueryIntegration.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        parent::setUp();
+
+        $this->auditErrorLoader = $this->get('akeneo_connectivity.connection.fixtures.audit_error_loader');
+        $this->selectErrorCountPerConnectionQuery = $this->get('akeneo_connectivity_connection.persistence.query.select_error_count_per_connection');
+    }
+
+    public function test_it_gets_error_count_per_connection()
+    {
+        $this->createHourlyErrorCounts([
+            ['sap', ErrorTypes::TECHNICAL, '2020-01-01 12:00:00', 5],
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalSelectErrorCountPerConnectionQueryIntegration.php:41]: The Technical error should be inside the expected date interval to check we don't select it.
```suggestion
            ['sap', ErrorTypes::TECHNICAL, '2020-01-02 12:00:00', 5],
```

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Persistence/Query/SelectErrorCountPerConnectionQuery.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+interface SelectErrorCountPerConnectionQuery
+{
+    /**
+     * @return ErrorCountPerConnection
+     */
+    public function execute(
+        string $eventType,
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Audit/Persistence/Query/SelectErrorCountPerConnectionQuery.php:20]: You must type int ErrorType here to validate the argument. I checked the code and you never validate it during the Query/Handler stack.

**Pull request merged**